### PR TITLE
feat: add api version to requests and responses FS-481

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-481" title="FS-481" target="_blank">Sub-taskFS-481</a>  [iOS] Pair requests and responses with api version
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With API versioning, all requests are made to a versioned endpoint. In order to correctly process the response of a versioned request, we must also be able to identify the api version of the response.

This PR adds an api version property to both `ZMTransportRequest` and `ZMTransportResponse`.

### Testing

No new tests have been added, but an assertion is made that when a request is completed with a response, the api versions must match.

### Notes 

For now we only consider that there exists version 0. In later work, we will start to support additional versions.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by sb88k — APPROVED** [review]
Is the logic for adding the version in the path included in this PR?
or are you adding the property only?

**Comment by johnxnguyen** [comment]
@sb88k adding the property only.

File: Source/Public/ZMTransportRequest.h [hunk 1]
```diff
@@ -126,6 +127,13 @@ typedef NS_ENUM(int8_t, ZMTransportAccept) {
 @property (nonatomic) BOOL shouldFailInsteadOfRetry;
 @property (nonatomic) BOOL doesNotFollowRedirects;
 
+/// The api version for which this request was made against.
+///
+/// In order to correctly handle the response to this request, the api version must
+/// be taking into account. Each request is paired with a response, so the api version
```
Comment (David-Henner) [inline Source/Public/ZMTransportRequest.h:133]:
```suggestion
/// be taken into account. Each request is paired with a response, so the api version
```

File: Source/Public/ZMTransportResponse.h [hunk 2]
```diff
@@ -68,6 +69,13 @@ typedef NS_ENUM(uint8_t, ZMTransportResponseStatus) {
 - (nullable NSString *)payloadLabel;
 
 @property (nonatomic, nullable) NSDate *startOfUploadTimestamp;
+
+/// The api version for associated with the response.
```
Comment (David-Henner) [inline Source/Public/ZMTransportResponse.h:73]:
```suggestion
/// The api version associated with the response.
```

**Review by David-Henner — APPROVED** [review]

**Pull request merged**